src/lib/updating-element.ts

```diff
@@ -15,72 +15,87 @@
 /**
  * Converts property values to and from attribute values.
  */
-type AttributeProcessor = {
+interface AttributeSerializer<T = any> {
+
   /**
    * Deserializing function called to convert an attribute value to a property value.
    */
-  fromAttribute?(value: string): unknown,
+  fromAttribute?(value: string): T;
+
   /**
    * Serializing function called to convert a property value to an attribute value.
    */
-  toAttribute?(value: unknown): string|null
-};
+  toAttribute?(value: T): string|null;
+
+}
+
+type AttributeType<T = any> = AttributeSerializer<T>|((value: string) => T);
 
 /**
  * Defines options for a property accessor.
  */
-export interface PropertyOptions {
+export interface PropertyDeclaration<T = any> {
+
   /**
-   * Describes how and if the property should becomes an observedAttribute.
-   * If explicitly false, the property will not be observed. Otherwise if true
-   * or absent, the lowercased property name is observed, and if a string
-   * observe that value.
+   * Describes how and if the property becomes an observedAttribute.
```

Comment:
I would just say "observed attribute"  - "observedAttribute" is not really a thing and you explain exactly what you mean below "is added to `observedAttributes`"

Reply:
but in a way you don't add things to `observedAttributes` but it is added to the array of strings that that getter returns

Reply:
Fixed